### PR TITLE
Validation feature

### PR DESCRIPTION
**Pull request by david-lafontant**
**Implement a simple validation:**

- The content of the email field has to be in lower case.

Implement the following interactions:

- [x] When the user submits the form, you check if the email is in lower case.
- [x] If the validation is OK, the form is sent.
- [x] If the validation is not OK, you show an error message to the user near the submit button informing them of the error and the form is not sent.

### CODE REVIEW

File: script/validation.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+function hasValue(input, message) {
+    if (input.value.trim() === "") {
+        return showError(input, message);
+    }
+
+return showSuccess(input);
+}
+
+function validateEmail(input, requiredMsg, invalidMsg) {
```
Comment (elyor-doniyorov) [inline script/validation.js:35]: - [ ] Make sure if a user submits the contact form with the email field containing `uppercase `characters an error message should show up. Kindly, fix it!

**Review by elyor-doniyorov — CHANGES_REQUESTED** [review]
Hi @david-lafontant, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost-there


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: index.html [hunk 2]
```diff
@@ -300,14 +300,15 @@ <h2 class="intro-heading form-page-heading">Contact me</h2>
...
         </section>
         <section class="form-page-section">
-          <form class="form-element" action="https://formspree.io/f/meqnjrqo" method="POST">
-            <label for="user-name"></label>
-            <input type="text" name="user-name" class="input-field text-input" placeholder="Please enter your name" id="user-name" maxlength="30" required />
-            <label for="user-email"></label>
-            <input type="email" name="-replyto" id="user-email" class="input-field text-input" placeholder="Please enter your email" pattern="[^@\s]+@[^@\s]+\.[^@\s]+" required />
+          <form class="form-element" action="https://formspree.io/f/meqnjrqo" method="POST" id="form">
+            <label for="name"></label>
+            <input type="text" name="name" class="input-field text-input" placeholder="Please enter your name" id="name" maxlength="30"/>
```
[the remark this answers is not in the document]
Comment (david-lafontant) [reply index.html]: Thank you for your comment. I implemented a solution for this issue.
[ solution on index.html](https://github.com/david-lafontant/Portfolio/blob/4f83af3ced89e4443414590177158f0b053600bf/index.html#L305)
[solution on the script](https://github.com/david-lafontant/Portfolio/blob/4f83af3ced89e4443414590177158f0b053600bf/script/validation.js#L68)

File: script/validation.js (hunk 1 above)
Comment (david-lafontant) [reply script/validation.js:35]: Thank you very much for your comment. I implemented a solution with regular expression in the validateEmail function in the script file. I hope this meets your expectations:
Link to the solution: [click here](https://github.com/david-lafontant/Portfolio/blob/4f83af3ced89e4443414590177158f0b053600bf/script/validation.js#L42)
validation-error-uppercase

File: script/validation.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    // Validate the form
+    const nameValid = hasValue(form.elements.name, NAME_REQUIRED);
+    const emailValid = validateEmail(form.elements.email, EMAIL_REQUIRED, EMAIL_INVALID);
+    // If valid, submit the form.
+
+    if (nameValid && emailValid) {
+    console.log('form event submit')
+      form.submit();
+    }
+  });
```
Comment (andersonlebon) [inline script/validation.js:76]: - [ ] As the previous review said. The user should be prevented from submitting the form with an email input containing an uppercase letter and a message should be displayed near the submit button. Let's go and implement this feature :muscle: :muscle: :muscle:

**Review by andersonlebon — CHANGES_REQUESTED** [review]
### Hi @david-lafontant 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## Suggested changes
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by alonsofl — APPROVED** [review]
Hi @david-lafontant,

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**